blocks/library/image/style.scss

```diff
@@ -30,13 +41,39 @@
 		.editor-block-mover {
 			display: none;
 		}
+	}
 
+	&[data-align="full"],
+	&[data-align="wide"] {
 		.editor-visual-editor__block-controls {
-			max-width: #{ $visual-editor-max-width - $block-padding - ( $block-padding * 2 + $block-mover-margin ) };
-			margin-left: auto;
-			margin-right: auto;
-		}
+			max-width: $visual-editor-max-width - ( ( $block-padding + $block-padding + $block-mover-margin ) * 2 );;
+			// a block is 600px wide inside a 700px container
+			// with block mover margin, border widths and paddings, toolbars are indented 46px
+			// that's 2px border plus 10px padding plus 18px mover margin
+			// these 46px are accounted for on both sides, to make sure the block is centered
+
+			width: auto;
+			margin-left: $block-mover-padding-hidden + $block-padding;
+			margin-right: $block-mover-padding-hidden + $block-padding;
+
+			@include break-small() {
+				width: 100%;
```

Comment:
The position of the controls isn't quite right when at wide or full alignments. You can see it move left and right when toggling between an image "Align left" and "Align wide".

I think we need to assign `width: $visual-editor-max-width; max-width: none;`